File: sdks/python/src/echo_agent_sdk/command_cell_status_values.py
from __future__ import annotations

from enum import Enum


class CommandCellTerminalCause(str, Enum):
    EXITED = "exited"
    TIMED_OUT = "timed_out"
    CANCELLED = "cancelled"
    LAUNCH_FAILED = "launch_failed"
    WAIT_FAILED = "wait_failed"
    OUTPUT_DRAIN_FAILED = "output_drain_failed"

    def as_str(self) -> str:
        return self.value


class CommandCellArtifactStatus(str, Enum):
    NOT_REQUESTED = "not_requested"
    WRITING = "writing"
    BELOW_THRESHOLD = "below_threshold"
    AVAILABLE = "available"
    FAILED = "failed"

    def as_str(self) -> str:
        return self.value

File: sdks/typescript/src/command_cell_status_values.ts
/** Command-cell terminal/artifact status values projected without ownership. */
export const CommandCellTerminalCause = Object.freeze({
  Exited: "exited",
  TimedOut: "timed_out",
  Cancelled: "cancelled",
  LaunchFailed: "launch_failed",
  WaitFailed: "wait_failed",
  OutputDrainFailed: "output_drain_failed",
} as const);

export type CommandCellTerminalCause =
  (typeof CommandCellTerminalCause)[keyof typeof CommandCellTerminalCause];

export function commandCellTerminalCauseAsStr(cause: CommandCellTerminalCause): string {
  if (!Object.values(CommandCellTerminalCause).includes(cause)) {
    throw new TypeError("invalid command cell terminal cause");
  }
  return cause;
}

export const CommandCellArtifactStatus = Object.freeze({
  NotRequested: "not_requested",
  Writing: "writing",
  BelowThreshold: "below_threshold",
  Available: "available",
  Failed: "failed",
} as const);

export type CommandCellArtifactStatus =
  (typeof CommandCellArtifactStatus)[keyof typeof CommandCellArtifactStatus];

export function commandCellArtifactStatusAsStr(status: CommandCellArtifactStatus): string {
  if (!Object.values(CommandCellArtifactStatus).includes(status)) {
    throw new TypeError("invalid command cell artifact status");
  }
  return status;
}
